includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -763,9 +765,26 @@ public function ajax_get_shipping_options() {
 				throw new Exception( __( 'Unable to find shipping method for address.', 'woocommerce-gateway-stripe' ) );
 			}
 
-			if ( isset( $data[0] ) ) {
+			// Check if at least single shipping option is available for new address.
```

Comment:
The comment here should include more about why wee need the funky sorting logic.  What are we trying to achieve? It will not be clear from the onset.

Reply:
Good point. Added better comment in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1191/commits/cdbaae722c3121c8208d7ef813cf4f4e033ae8af.